tests/unit_tests/device/cuda/detector_cuda.cpp

```diff
@@ -96,3 +99,81 @@ TEST(detector_cuda, detector) {
+    // the device
+    // use synchronous copy and fixed size buffers
+    auto det_buff_static = detray::get_buffer(det_host, dev_mr, cuda_cpy);
+
+    // ---------- construct an "aligned" transform store ---------
+    // a few typedefs
+    using test_algebra = test::algebra;
+    using scalar = dscalar<test_algebra>;
+    using point3 = dpoint3D<test_algebra>;
+
+    // build a vector of aligned transforms on the host
+    // for populating this vector take all transforms of the detector
+    // and shift them by the same translation
+    typename detector_host_t::transform_container tf_store_aligned_host;
+
+    point3 shift{.1f * unit<scalar>::mm, .2f * unit<scalar>::mm,
+                 .3f * unit<scalar>::mm};
+
+    tf_store_aligned_host.reserve(
+        det_host.transform_store().size(),
+        typename decltype(det_host)::transform_container::context_type{});
+
+    for (const auto& tf : det_host.transform_store()) {
+        point3 shifted = tf.translation() + shift;
+        tf_store_aligned_host.push_back(
+            transform_t{shifted, tf.x(), tf.y(), tf.z()});
+    }
+
+    // copy the vector of aligned transforms to the device
+    // again, use synchronous copy and fixed size buffers
+    auto tf_buff_aligned =
+        get_buffer(tf_store_aligned_host, dev_mr, cuda_cpy, copy::sync,
+                   vecmem::data::buffer_type::fixed_size);
+
+    // Get the view of the aligned detector using the vector of aligned
+    // transforms and the static part of the detector copied to the device
+    // earlier
+    auto detector_view_aligned =
+        detail::misaligned_detector_view<detector_host_t>(det_buff_static,
+                                                          tf_buff_aligned);
+    // Get the view of the static detector
+    auto detector_view_static = detray::get_data(det_buff_static);
+
+    // make two vectors for surface transforms copied from device side
+    vecmem::vector<transform_t> surfacexf_device_static(
+        det_host.surfaces().size(), &mng_mr);
+    vecmem::vector<transform_t> surfacexf_device_aligned(
+        det_host.surfaces().size(), &mng_mr);
+    // views of the above vectors
+    auto surfacexf_data_static = vecmem::get_data(surfacexf_device_static);
+    auto surfacexf_data_aligned = vecmem::get_data(surfacexf_device_aligned);
+
+    // run the test code to extract the surface transforms for the static
+    // and misaligned detector views and to store them into the vectors
+    detector_alignment_test(detector_view_static, detector_view_aligned,
+                            surfacexf_data_static, surfacexf_data_aligned);
+
+    // check that the relevant transforms have been properly shifted
+    for (unsigned int i = 0u; i < surfacexf_device_static.size(); i++) {
+        auto transdiff = surfacexf_device_aligned[i].translation() -
+                         surfacexf_device_static[i].translation();
+        EXPECT_EQ(transdiff == shift, true);
```

Comment:
Does this account for potential floating point differences? We have a new macro that might help here:
tests/include/detray/test/common/assert.hpp

Reply:
Thanks. I switched to `EXPECT_POINT3_NEAR` in the latest revision.

I was also puzzled by the failure of the Eigen test in CI. The failure was reproducible in my local runs and after playing around with it for a while, here is what made it work for me:
```
@@ -174,8 +174,9 @@ TEST(detector_cuda, detector_alignment) {
 
     // check that the relevant transforms have been properly shifted
     for (unsigned int i = 0u; i < surfacexf_device_static.size(); i++) {
-        auto transdiff = surfacexf_device_aligned[i].translation() -
-                         surfacexf_device_static[i].translation();
-        EXPECT_POINT3_NEAR(transdiff, shift, 1e-6);
+        auto translation_static = surfacexf_device_static[i].translation();
+        auto translation_aligned = surfacexf_device_aligned[i].translation();
+        auto translation_diff = translation_aligned - translation_static;
+        EXPECT_POINT3_NEAR(translation_diff, shift, 1e-6);
     }
 }
```
Otherwise the `transdiff` was always = `{0,0,0}`. I have no explanation for this ......

Reply:
Yes, that is indeed curious. Sometimes it helps using explicit types with Eigen instead of ```auto```. But if it works now, let's leave it like it is.